### PR TITLE
14429 Misc filing survey fixes

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* bcgov/entity#14429

*Description of changes:*
- misc filing survey fixes (see PR comments for details)
- app version = 4.5.9

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -417,6 +418,12 @@ export default class App extends Vue {
     return !this.$route.meta.noStepper
   }
 
+  /** The current domain. */
+  get domain (): string {
+    // remove possible leading period
+    return window.location.hostname.replace(/^\./, '')
+  }
```
Comment (severinbeauvais) [inline src/App.vue:425]: This value is used 3 times in this component, and needed some extra logic to handle some domains, for example:
- `localhost`
- `.business-create-dev.apps.silver.devops.gov.bc.ca`

File: src/App.vue [hunk 2]
```diff
@@ -1106,6 +1102,19 @@ export default class App extends Vue {
...
+        // - if the IA Survey ID is configured
+        // - if the cookie doesn't exist (eg, never set, or expired)
+        if (this.isIncorporationFiling) {
+          if (sessionStorage.getItem('IA_SURVEY_ID')) {
+            const cookie = Cookies.get(this.IA_SURVEY_KEY, { domain: this.domain })
+            if (!cookie) {
+              this.filingSurveyDialog = true
+            }
+          }
+        }
```
Comment (severinbeauvais) [inline src/App.vue:1117]: Moved here as this block is called only once, whereas fetchData() (where this was before) could be called additional times to retry from an error dialog.

File: src/utils/FetchConfig.ts [hunk 3]
```diff
@@ -95,7 +95,8 @@ export async function FetchConfig (): Promise<any> {
   }
 
   const iaSurveyId: string = response.data.IA_SURVEY_ID
-  sessionStorage.setItem('IA_SURVEY_ID', iaSurveyId)
+  // NB: assign fallback otherwise sessionStorage will store string "undefined"
+  sessionStorage.setItem('IA_SURVEY_ID', iaSurveyId || '')
```
Comment (severinbeauvais) [inline src/utils/FetchConfig.ts:99]: `sessionStorage.setItem('IA_SURVEY_ID', null)` will save a string value of "null".

`sessionStorage.setItem('IA_SURVEY_ID', undefined)` will save a string value of "undefined".

So if you want a falsy value in there, you need to set empty string.

**Review by jonathan-longe — APPROVED** [review]
LGTM

**Review by JazzarKarim — APPROVED** [review]

**Review by ChrisGabel23 — APPROVED** [review]

**Pull request merged**